packages/bot-web-ui/src/stores/load-modal-store.js

```diff
@@ -0,0 +1,248 @@
+        this.active_index = index;
+
+        if (this.active_index !== 0 &&
+            this.recent_workspace &&
+            this.recent_workspace.rendered
+        ) {
+            this.recent_workspace.dispose();
+        }
+        if (this.active_index === 0 && this.recent_files.length) {
+            this.previewWorkspace({ id: this.selected_file });
+        }
+        if (this.active_index !== 1 &&
+            this.loaded_local_file &&
+            this.local_workspace &&
+            this.local_workspace.rendered
+        ) {
+            this.local_workspace.dispose();
+            this.loaded_local_file = null;
+        }
+        if (this.active_index === 1) {
+            this.drop_zone = document.getElementsByClassName('local__dragndrop')[0];
+            this.drop_zone.addEventListener('drop', e => this.handleFileChange(e, false));
+        } else if (this.drop_zone) {
+            this.drop_zone.removeEventListener('drop', e => this.handleFileChange(e, false));
+        }
+    }
+
+    /** --------- Recent Tab Start --------- */
+    @action.bound
+    onMount() {
+        if (this.recent_files.length && this.active_index === 0) {
+            this.selected_file = this.recent_files[0].id;
+            this.previewWorkspace({ id: this.selected_file });
+        }
+    }
+
+    @action.bound
+    onUnmount() {
+        if (this.recent_workspace && this.recent_workspace.rendered) {
+            this.recent_workspace.dispose();
+        }
+        this.selected_file = null;
+        this.setActiveTabIndex(0);
+    }
+
+    @action.bound
+    previewWorkspace({ id }) {
+        const selected_file = this.recent_files.find(file => file.id === id);
+        if (!selected_file) {
+            return;
+        }
+
+        const xml_file = selected_file.xml;
+        this.selected_file = id;
+        
+        if (!this.recent_workspace || !this.recent_workspace.rendered) {
+            const ref = document.getElementById('scratch_recent');
+            this.recent_workspace   = Blockly.inject(ref, {
+                media   : `${__webpack_public_path__}media/`, // eslint-disable-line
+                zoom : {
+                    wheel     : false,
+                    startScale: config.workspaces.previewWorkspaceStartScale,
+                },
+                readOnly: true,
+            });
+        } else {
+            this.recent_workspace.clear();
+        }
+
+        Blockly.Xml.domToWorkspace(Blockly.Xml.textToDom(xml_file), this.recent_workspace);
+    }
+
+    @action.bound
+    onZoomInOutClick(is_zoom_in) {
+        const metrics   = this.recent_workspace.getMetrics();
+        const addition  = is_zoom_in ? 1 : -1;
+
+        let workspace;
+        if (this.active_index === 0) {
+            workspace = this.recent_workspace;
+        } else if (this.active_index === 1) {
+            workspace = this.local_workspace;
+        }
+
+        workspace.zoom(metrics.viewWidth / 2, metrics.viewHeight / 2, addition);
```

Comment:
You can use `workspace.zoomCenter(type)` instead. e.g.

```suggestion
        workspace.zoomCenter(is_zoom_in ? 1 : 0)
```